### PR TITLE
Move the device ID to settings and linter cleanups closes #4319

### LINKED ISSUE #4319: Make device ID accessible via settings
devel and pro diverged in this regard, and it currently causes the pro UI to not load.

### PR DESCRIPTION
**Pull request by myleshorton**

### CODE REVIEW

File: src/github.com/getlantern/flashlight/app/app.go [hunk 1]
```diff
@@ -21,7 +21,8 @@ import (
 )
 
 var (
-	log = golog.LoggerFor("flashlight")
+	log      = golog.LoggerFor("flashlight")
+	settings *Settings
```
Comment (myleshorton) [inline src/github.com/getlantern/flashlight/app/app.go]: The settings variable used to have issues with shadowing.

File: src/github.com/getlantern/flashlight/app/app.go [hunk 2]
```diff
@@ -52,20 +53,11 @@ func (app *App) Init() {
...
-	)
-	if err != nil {
-		panic("Error initializing config")
-	}
 	if err := logging.EnableFileLogging(); err != nil {
 		panic("Error initializing logging")
 	}
 
-	<-logging.Configure("", cfg.Client.DeviceID, flashlight.Version, flashlight.RevisionDate)
+	<-logging.Configure("", "dummy-device-id-for-panic", flashlight.Version, flashlight.RevisionDate)
```
Comment (myleshorton) [inline src/github.com/getlantern/flashlight/app/app.go]: We don't need the device ID when reporting a panic, and the previous code went out of its way to get a config just for that purpose.

File: src/github.com/getlantern/flashlight/app/settings.go [hunk 3]
```diff
@@ -49,11 +51,10 @@ func loadSettings(version, revisionDate, buildDate string) *Settings {
 
 // loadSettings loads the initial settings at startup, either from disk or using defaults.
 func loadSettingsFrom(version, revisionDate, buildDate, path string) *Settings {
-
 	log.Debug("Loading settings")
 	// Create default settings that may or may not be overridden from an existing file
 	// on disk.
-	settings = &Settings{
+	set := &Settings{
```
Comment (myleshorton) [inline src/github.com/getlantern/flashlight/app/settings.go]: Just keep this variable local now to avoid shadowing issues and generally confusing code.

File: src/github.com/getlantern/flashlight/client/client.go [hunk 4]
```diff
@@ -55,12 +55,18 @@ type Client struct {
 	rp eventual.Value
 
 	l net.Listener
+
+	proxyAll func() bool
```
Comment (myleshorton) [inline src/github.com/getlantern/flashlight/client/client.go]: Just store the proxyAll function in the struct instead of re-configuring the value every time the config changes. Eliminates threading issues as well as simplifies the code.

File: src/github.com/getlantern/flashlight/client/client.go [hunk 5]
```diff
@@ -166,10 +178,8 @@ func (client *Client) Configure(cfg *ClientConfig, proxyAll func() bool) {
 
 	log.Debugf("Requiring minimum QOS of %d", cfg.MinQOS)
 	client.cfgHolder.Store(cfg)
-	log.Debugf("Proxy all traffic or not: %v", proxyAll())
-	client.proxyAll.Store(proxyAll)
```
Comment (myleshorton) [inline src/github.com/getlantern/flashlight/client/client.go]: Don't need to store this atomically any more, as it never changes.

File: src/github.com/getlantern/flashlight/client/handler.go [hunk 6]
```diff
@@ -70,13 +70,13 @@ func (client *Client) intercept(resp http.ResponseWriter, req *http.Request) {
 	// after completed send / receive so that won't cause problem.
 	closeConns := func() {
 		if clientConn != nil {
-			if err := clientConn.Close(); err != nil {
-				log.Debugf("Error closing the out connection: %s", err)
+			if closeErr := clientConn.Close(); closeErr != nil {
```
Comment (myleshorton) [inline src/github.com/getlantern/flashlight/client/handler.go]: Linter

File: src/github.com/getlantern/flashlight/client/handler.go [hunk 7]
```diff
@@ -70,13 +70,13 @@ func (client *Client) intercept(resp http.ResponseWriter, req *http.Request) {
...
+				log.Debugf("Error closing the out connection: %s", closeErr)
 			}
 		}
 		if connOut != nil {
-			if err := connOut.Close(); err != nil {
-				log.Debugf("Error closing the client connection: %s", err)
+			if closeErr := connOut.Close(); closeErr != nil {
```
Comment (myleshorton) [inline src/github.com/getlantern/flashlight/client/handler.go]: Linter

File: src/github.com/getlantern/flashlight/client/pluggable_transports.go [hunk 8]
```diff
@@ -53,8 +53,8 @@ func defaultDialFactory(s *ChainedServerInfo, deviceID string) (dialFN, error) {
 				return nil, err
 			}
 			if !forceProxy && !conn.ConnectionState().PeerCertificates[0].Equal(x509cert) {
-				if err := conn.Close(); err != nil {
-					log.Debugf("Error closing chained server connection: %s", err)
+				if closeErr := conn.Close(); closeErr != nil {
```
Comment (myleshorton) [inline src/github.com/getlantern/flashlight/client/pluggable_transports.go]: Linter

File: src/github.com/getlantern/lantern/lantern.go [hunk 9]
```diff
@@ -127,6 +130,7 @@ func run(configDir string) {
 		func(cfg *config.Config) {},                   // onConfigUpdate
 		&userConfig{},
 		func(err error) {}, // onError
+		base64.StdEncoding.EncodeToString(uuid.NodeID()),
```
Comment (myleshorton) [inline src/github.com/getlantern/lantern/lantern.go]: Just generate the device ID here on mobile...actually tempting to do something like that on desktop but the current should work fine...

**Comment by myleshorton** [comment]
Hey @xiam you mind giving this a look?

File: src/github.com/getlantern/flashlight/config/config.go [hunk 10]
```diff
@@ -227,8 +224,6 @@ func (updated *Config) applyFlags(flags map[string]interface{}) error {
 			updated.CloudConfigCA = value.(string)
 		case "frontedconfig":
 			updated.FrontedCloudConfig = value.(string)
-		case "instanceid":
```
Comment (xiam) [inline src/github.com/getlantern/flashlight/config/config.go]: Why was this removed?

File: src/github.com/getlantern/lantern/lantern.go (hunk 9 above)
Comment (atavism) [inline src/github.com/getlantern/lantern/lantern.go]: Nice! That's actually a lot cleaner. Do we also need to update ``GetToken()`` to return this?

File: src/github.com/getlantern/flashlight/config/config.go (hunk 10 above)
Comment (myleshorton) [inline src/github.com/getlantern/flashlight/config/config.go]: Because it set the device ID in the ClientConfig, which doesn't exist anymore. That's the one thing we lose with this change -- the server can no longer set the device ID. My thought was to add that if we ever need it @xiam.

File: src/github.com/getlantern/lantern/lantern.go (hunk 9 above)
Comment (myleshorton) [inline src/github.com/getlantern/lantern/lantern.go]: It's actually just the device ID and not the pro token @atavism

**Comment by xiam** [comment]
Looks good @myleshorton, thanks!

**Comment by myleshorton** [comment]
Awesome @xiam thanks!
